cartofacade/capi_test.go

```diff
@@ -113,59 +154,80 @@ func TestCGoAPI(t *testing.T) {
+		// second, the map doesn't update (at least not after 10 lidar readings).
+		// I'm not sure why cartotgrapher has this behavior.
+		// Cartographer does not provide any feedback regarding
+		// why or when it does / does not update the map.
+		// As a result, these tests show best case behavior.
 
-		// test valid addSensorReading: valid reading ascii
-		err = pointcloud.ToPCD(pc, buf, 0)
-		test.That(t, err, test.ShouldBeNil)
-		err = vc.addSensorReading("mysensor", buf.Bytes(), timestamp)
+		// 1. test valid addSensorReading: valid reading ascii
+		t.Log("sensor reading 1")
+		timestamp = timestamp.Add(time.Second * 2)
+		testAddSensorReading(t, vc, "mysensor", "viam-cartographer/mock_lidar/0.pcd", timestamp, pointcloud.PCDAscii)
+
+		// test getPosition zeroed if not enough sensor data has been provided
+		position, err := vc.getPosition()
 		test.That(t, err, test.ShouldBeNil)
+		test.That(t, position.ComponentReference, test.ShouldEqual, "mysensor")
+		positionIsZero(t, position)
 
-		// confirm the pointcloud package still doesn't support binary compressed
-		// pointclouds. If it does, we need to implement:
-		// https://viam.atlassian.net/browse/RSDK-3753
-		err = pointcloud.ToPCD(pc, buf, 2)
+		// test getPointCloudMap returns error if not enough sensor data has been provided
+		pcd, err := vc.getPointCloudMap()
+		test.That(t, pcd, test.ShouldBeNil)
 		test.That(t, err, test.ShouldBeError)
-		test.That(t, err.Error(), test.ShouldResemble, "compressed PCD not yet implemented")
+		test.That(t, err, test.ShouldResemble, errors.New("VIAM_CARTO_POINTCLOUD_MAP_EMPTY"))
 
-		// test getPosition
-		holder, err := vc.getPosition()
+		// 2. test valid addSensorReading: valid reading binary
+		t.Log("sensor reading 2")
+		timestamp = timestamp.Add(time.Second * 2)
+		testAddSensorReading(t, vc, "mysensor", "viam-cartographer/mock_lidar/1.pcd", timestamp, pointcloud.PCDBinary)
 
+		// test getPosition zeroed
+		position, err = vc.getPosition()
 		test.That(t, err, test.ShouldBeNil)
-		test.That(t, holder.ComponentReference, test.ShouldEqual, "mysensor")
+		test.That(t, position.ComponentReference, test.ShouldEqual, "mysensor")
+		positionIsZero(t, position)
 
-		test.That(t, holder.X, test.ShouldEqual, 0)
-		test.That(t, holder.Y, test.ShouldEqual, 0)
-		test.That(t, holder.Z, test.ShouldEqual, 0)
+		// test getPointCloudMap now returns a non empty result
+		pcd, err = vc.getPointCloudMap()
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, pcd, test.ShouldNotBeNil)
+		pc, err := pointcloud.ReadPCD(bytes.NewReader(pcd))
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, pc.Size(), test.ShouldNotEqual, 0)
 
-		test.That(t, holder.Imag, test.ShouldEqual, 0)
-		test.That(t, holder.Jmag, test.ShouldEqual, 0)
-		test.That(t, holder.Kmag, test.ShouldEqual, 0)
-		test.That(t, holder.Real, test.ShouldEqual, 1)
+		// third sensor reading populates the pointcloud map and the  position
```

Comment:
typo: extra space between `the` and `position`

Reply:
done